server/api.js

```diff
@@ -547,6 +554,61 @@ router.post("/bid/:bidId/status", async (req, res) => {
 	}
 });
 
+router.post("/bid", upload.single("document"), async (req, res) => {
+	try {
+		const { tenderId, bidding_amount, cover_letter, suggested_duration_days } =
+			req.body;
+		const { originalname } = req.file;
+		const bidderId = req.user.id;
+		const biddingDate = new Date();
+		const status = "Pending";
+
+		const client = await pool.connect();
+
+		try {
+			await client.query("BEGIN");
+
+			const bidQuery = `
+				INSERT INTO bid (tender_id, bidder_id, bidding_date, status, bidding_amount, cover_letter, suggested_duration_days)
+				VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING bid_id
+			`;
+
+			const bidValues = [
+				tenderId,
+				bidderId,
+				biddingDate,
+				status,
+				bidding_amount,
+				cover_letter,
+				suggested_duration_days,
+			];
+
+			const bidResult = await client.query(bidQuery, bidValues);
+			const bidId = bidResult.rows[0].bid_id;
+
+			const attachmentQuery = `
+				INSERT INTO bid_attachment (bid_id, attachment)
+				VALUES ($1, $2)
+			`;
+
+			const attachmentValues = [bidId, originalname];
```

Comment:
You are saving the attachment name to the database, but the file is not being saved anywhere.

Saving the file it's going to be a little more work, let's remove it from here and add it in another commit.